azurerm/internal/services/logic/logic_app_workflow_resource.go

```diff
@@ -290,6 +303,10 @@ func resourceArmLogicAppWorkflowRead(d *schema.ResourceData, meta interface{}) e
 			}
 		}
 
+		if props.IntegrationServiceEnvironment != nil && props.IntegrationServiceEnvironment.ID != nil {
+			d.Set("integration_service_environment_id", props.IntegrationServiceEnvironment.ID)
+		}
```

Comment:
we need to ensure this is always set into the state - so can we make this:

```suggestion
		integrationServiceEnvironmentId := ""
		if props.IntegrationServiceEnvironment != nil && props.IntegrationServiceEnvironment.ID != nil {
 			integrationServiceEnvironmentId := *props.IntegrationServiceEnvironment.ID
 		}
 		d.Set("integration_service_environment_id", integrationServiceEnvironmentId)
```